ttsfm/client.py

```
@@ -0,0 +1,409 @@
+                        raise exception
+                    
+                    logger.warning(f"Request failed with status {response.status_code}, retrying...")
+                    continue
+                    
+            except requests.exceptions.Timeout:
+                if attempt == self.max_retries:
+                    raise NetworkException(
+                        f"Request timed out after {self.timeout}s",
+                        timeout=self.timeout,
+                        retry_count=attempt
+                    )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)